plugins/woocommerce/src/Blocks/BlockTypes/AbstractBlock.php

```diff
@@ -151,11 +151,14 @@ protected function register_block_type_assets() {
 			$data     = $this->asset_api->get_script_data( $this->get_block_type_script( 'path' ) );
 			$has_i18n = in_array( 'wp-i18n', $data['dependencies'], true );
 
+			$has_blocks_vendors = ! strpos( $this->get_block_type_script( 'handle' ), 'checkout' ) && ! strpos( $this->get_block_type_script( 'handle' ), 'cart' );
```

Comment:
This is a bit sus, adding vendor deps can be quite manual. I'd love to build some concept of the bundle in PHP rather than what we have in AssetsController right now but it'd be quite a refactor.

We need to ensure that only non cart and checkout blocks load this vendor script, we could use class inheritance instead somehow? ie we could include this by default in AbstractBlock but have  `Cart.php` and `Checkout.php` inherit from a class that has it turned off, but it's introducing a lot of class stuff just to turn off or on a vendor script.

I've also considered whether we should submit a PR upstream to dependency extraction webpack plugin, my idea is that it would be useful to specify additional dependencies when generating the php asset file.